src/overlays/actors/ovl_Bg_Spout_Fire/z_bg_spout_fire.c

```diff
@@ -31,39 +34,172 @@ const ActorInit Bg_Spout_Fire_InitVars = {
+    }
+}
+
+void func_80A60CDC(BgSpoutFire* this, GlobalContext* globalCtx) {
+    if (this->unk_148 != 0) {
+        this->unk_148--;
+    }
+    if (this->unk_148 == 0) {
+        this->actionFunc = func_80A60D10;
+    }
+}
+
+void func_80A60D10(BgSpoutFire* this, GlobalContext* globalCtx) {
+    if (func_80A60C24(this, globalCtx)) {
+        Math_StepToF(&this->actor.scale.y, 0.1f, 0.024999999f);
+        return;
+    }
+    if (Math_StepToF(&this->actor.scale.y, 0.01f, 0.01f)) {
+        this->actor.draw = NULL;
+        this->actionFunc = func_80A60C94;
+        return;
+    }
+    this->unk_148 = 0;
+}
+
+void func_80A60DA0(Actor* thisx, GlobalContext* globalCtx) {
+    s16 phi_a3;
+
+    if (Actor_IsFacingPlayer(thisx, 0x4000)) {
+        phi_a3 = thisx->shape.rot.y;
+    } else {
+        phi_a3 = (thisx->shape.rot.y + 0x8000);
+    }
+    func_800B8D98(globalCtx, thisx, 5.0f, phi_a3, 1.0f);
+}
+
+void func_80A60E08(BgSpoutFire* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f sp30;
+    f32 cos;
+    f32 sin;
+
+    Actor_OffsetOfPointInActorCoords(&this->actor, &sp30, &player->actor.world.pos);
+    sp30.x = CLAMP(sp30.x, -74.25f, 74.25f);
+    if (this->unk_148 == 0) {
+        if (sp30.z > 0.0f) {
+            sp30.z = -25.0f;
+            this->unk_148 = -1;
+        } else {
+            sp30.z = 25.0f;
+            this->unk_148 = 1;
+        }
+    } else {
+        sp30.z = this->unk_148 * 25.0f;
+    }
+    sin = Math_SinS(this->actor.shape.rot.y);
+    cos = Math_CosS(this->actor.shape.rot.y);
+    this->unk14C.dim.pos.x = (this->actor.world.pos.x + (sp30.x * cos) + (sp30.z * sin));
+    this->unk14C.dim.pos.z = (this->actor.world.pos.z - (sp30.x * sin) + (sp30.z * cos));
+}
+
+void BgSpoutFire_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s16 pad;
+    BgSpoutFire* this = THIS;
+
+    this->unk_14A = ((this->unk_14A + 1) % 8);
+    if ((this->unk14C.base.atFlags & AT_HIT)) {
+        this->unk14C.base.atFlags &= ~AT_HIT;
+        func_80A60DA0(&this->actor, globalCtx);
+    }
+    this->actionFunc(this, globalCtx);
+    if (this->actionFunc == func_80A60D10) {
+        func_80A60E08(this, globalCtx);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->unk14C.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->unk14C.base);
+        func_800B9010(&this->actor, NA_SE_EV_FIRE_PLATE - SFX_FLAG);
+    }
+}
+
+void BgSpoutFire_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpoutFire* this = THIS;
+    Gfx* gfx;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    gfx = Gfx_CallSetupDL(POLY_XLU_DISP, 0x14);
+    POLY_XLU_DISP = gfx;
+    gSPSegment(&gfx[0], 0x08, D_80A61194[this->unk_14A]);
+    gDPSetPrimColor(&gfx[1], 0, 0x01, 255, 255, 0, 150);
+    gDPSetEnvColor(&gfx[2], 255, 0, 0, 255);
+    Matrix_Translate(-55.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    gSPMatrix(&gfx[3], Matrix_NewMtx(globalCtx->state.gfxCtx), (0x00 | 0x02) | 0x00);
```

Comment:
`gfxdis.f3dex2` should give the proper defines for the `(0x00 | 0x02) | 0x00`

Reply:
it didn't and i don't know what they should be